### PR TITLE
[v12] OpenTelemetry Update Backport

### PR DESCRIPTION
**Pull request by jentfoo**
v12 backport of PR https://github.com/gravitational/teleport/pull/33523 and https://github.com/gravitational/teleport/pull/33550

Similar to v13 backport #33551, we must take in the latest version of OpenTelemetry to address their http/2 https://github.com/advisories/GHSA-rcjv-mgp8-qvmr. As such this also increases the grpc version to match v14 and master

#21190 needed to also be backported in order to handle an incompatibility with Firestone

### CODE REVIEW

**Comment by jentfoo** [comment]
@xacrimon Can you help confirm we can backport https://github.com/gravitational/teleport/pull/21190 to v12

**Review by r0mant — APPROVED** [review]
Bot.

File: go.mod [hunk 1]
```diff
@@ -374,6 +367,12 @@ require (
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
 )
 
+require (
```
Comment (codingllama) [inline go.mod:370]: Merge with the require block above?

**Review by codingllama — APPROVED** [review]

File: api/go.mod [hunk 2]
```diff
@@ -12,45 +12,44 @@ require (
...
-	go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc v0.38.0
-	go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.39.0
-	go.opentelemetry.io/otel v1.13.0
-	go.opentelemetry.io/otel/exporters/otlp/otlptrace v1.12.0
-	go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc v1.12.0
-	go.opentelemetry.io/otel/sdk v1.12.0
-	go.opentelemetry.io/otel/trace v1.13.0
-	go.opentelemetry.io/proto/otlp v0.19.0
+	github.com/stretchr/testify v1.8.4
+	go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc v0.42.0
```
[the remark this answers is not in the document]
Comment (jentfoo) [reply api/go.mod]: Updated, thank you!

**Pull request merged**